### PR TITLE
Creating new apps

### PR DESCRIPTION
**Pull request by phamann**
Adds a new page to the developer guide with steps and troubleshooting to consider when creating a new applications within the stack.

/cc @wheresrhys @matthew-andrews

### CODE REVIEW

File: docs/developer-guide/creating-new-apps.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+section: Developer guide
+permalink: /docs/developer-guide/creating-new-apps/
+---
+
+# Creating new apps
+
+So you'd like to create a new app. First things first, assess whether or not you really need
+to create a new app: 
+
+- Could your endpoint/template/job exisit within an existing app?
```
Comment (matthew-andrews) [inline docs/developer-guide/creating-new-apps.md]: exisit

File: docs/developer-guide/creating-new-apps.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+# Creating new apps
+
+So you'd like to create a new app. First things first, assess whether or not you really need
+to create a new app: 
+
+- Could your endpoint/template/job exisit within an existing app? 
+- Does it really need it's own build system?
+- Does it really need it's own delpoyment process?
```
Comment (matthew-andrews) [inline docs/developer-guide/creating-new-apps.md]: delpyment

File: docs/developer-guide/creating-new-apps.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+- Create a new GitHub repo within the [Financial-Times](https://github.com/Financial-Times) organisation and ensure it is public.
+- Depending on our applications needs, copy a base directory structure and files from a similar existing application:
+	- If it's an api: [next-mustang](https://github.com/Financial-Times/next-mustang)
+	- If it has user facing views [next-engels](https://github.com/Financial-Times/next-engels)
+	-  *TODO:* create a base application for others to fork/copy from
+- Adjust the project `Makefile` to your needs 
+	- Does it need an `nbt build` step?
+	- Does it need an `nbt deploy-hased-assets` step?
+	- If not add the `--skip-layout-checks` to `nbt verify`
+- Ensure the app is correctly named within the `package.json` and `bower.json` maifests and have the correct dependencies listed.
```
Comment (matthew-andrews) [inline docs/developer-guide/creating-new-apps.md]: maifests

File: docs/developer-guide/creating-new-apps.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+- Make your changes and develop the initial app locally
+- Ensure `make run` and `make test` work appropriately.
+
+## Ready to deploy:
+ 
+- Add a new property to the [config-vars](http://git.svc.ft.com/projects/NEXTPRIVATE/repos/config-vars/browse/models/production.json) data including the enviornment vars your applicaiton needs.
+	- *Note:* The Sentry ARN and URL must be unique. You can create a new app in Sentry [here](https://app.getsentry.com/nextftcom/) to obtain an ARN.
+	- *Note:* We prefer each application to have it's own unique API key. Get a new one from the list [here](https://docs.google.com/spreadsheets/d/1y8jZMQUjHh8jyyOXMIedEkw03IRoxUIGy88d58782WY)
+	- Verify your change has deployed before moving on: `curl -I https://https://ft-next-config-vars.herokuapp.com/app/YOUR_APP_NAME?apiKey=YOUR_API_KEY`
```
Comment (matthew-andrews) [inline docs/developer-guide/creating-new-apps.md]: `https://https://`

File: docs/developer-guide/creating-new-apps.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	- Join the [ft-next-config-vars](https://dashboard.heroku.com/apps/ft-next-config-vars/resources) app in Heroku
+- Create a new application within Heroku correseponding to your apps name:
+	- `heroku create YOUR_APP_NAME --region eu --org financial-times`
+	- Test your apps configuration running `nbt configure`
+- Configure your Travis CI build via the dashboard:
+	- Go to: https://travis-ci.org/Financial-Times/YOUR_APP_GITHUB_NAME/settings 
+	- Click on envinorment variables tab and add the following defaults:
+		- `HEROKU_AUTH_TOKEN`, `GITHUB_AUTH_TOKEN` & `SPLUNK_URL`
+	- Add any application specfic env vars your appl may need to build.
```
Comment (matthew-andrews) [inline docs/developer-guide/creating-new-apps.md]: specfic

File: docs/developer-guide/creating-new-apps.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+- Does it really need it's own delpoyment process?
+
+If you are **sure** you still need to create a new application within the stack, follow the steps below:
+
+## Initialize the app:
+
+- Create a new GitHub repo within the [Financial-Times](https://github.com/Financial-Times) organisation and ensure it is public.
+- Depending on our applications needs, copy a base directory structure and files from a similar existing application:
+	- If it's an api: [next-mustang](https://github.com/Financial-Times/next-mustang)
+	- If it has user facing views [next-engels](https://github.com/Financial-Times/next-engels)
```
Comment (matthew-andrews) [inline docs/developer-guide/creating-new-apps.md]: i don't have a better alternative yet but it's sad we have to copy and paste :(

File: docs/developer-guide/creating-new-apps.md (hunk 2 above)
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: its

File: docs/developer-guide/creating-new-apps.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	- Does it need an `nbt build` step?
+	- Does it need an `nbt deploy-hased-assets` step?
+	- If not add the `--skip-layout-checks` to `nbt verify`
+- Ensure the app is correctly named within the `package.json` and `bower.json` maifests and have the correct dependencies listed.
+- Make your changes and develop the initial app locally
+- Ensure `make run` and `make test` work appropriately.
+
+## Ready to deploy:
+ 
+- Add a new property to the [config-vars](http://git.svc.ft.com/projects/NEXTPRIVATE/repos/config-vars/browse/models/production.json) data including the enviornment vars your applicaiton needs.
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: applicaiton/application

File: docs/developer-guide/creating-new-apps.md (hunk 5 above)
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: appl

File: docs/developer-guide/creating-new-apps.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+- Could your endpoint/template/job exist within an existing app? 
+- Does it really need its own build system?
+- Does it really need its own deployment process?
+
+If you are **sure** you still need to create a new application within the stack, follow the steps below:
+
+## Initialize the app:
+
+- Create a new GitHub repo within the [Financial-Times](https://github.com/Financial-Times) organisation and ensure it is public.
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: We're trying to stick with American English, so we should probably go with "organization", reluctantly

File: docs/developer-guide/creating-new-apps.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+- Depending on our applications needs, copy a base directory structure and files from a similar existing application:
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: applications/application's

File: docs/developer-guide/creating-new-apps.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+	- If its an api: [next-mustang](https://github.com/Financial-Times/next-mustang)
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: it's

File: docs/developer-guide/creating-new-apps.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+	- Does it need an `nbt build` step?
+	- Does it need an `nbt deploy-hased-assets` step?
+	- If not add the `--skip-layout-checks` to `nbt verify`
+- Ensure the app is correctly named within the `package.json` and `bower.json` manifests and have the correct dependencies listed.
+- Make your changes and develop the initial app locally
+- Ensure `make run` and `make test` work appropriately.
+
+## Ready to deploy:
+ 
+- Add a new property to the [config-vars](http://git.svc.ft.com/projects/NEXTPRIVATE/repos/config-vars/browse/models/production.json) data including the enviornment vars your application needs.
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: enviornment/environment

File: docs/developer-guide/creating-new-apps.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+	- *Note:* The Sentry ARN and URL must be unique. You can create a new app in Sentry [here](https://app.getsentry.com/nextftcom/) to obtain an ARN.
+	- *Note:* We prefer each application to have it's own unique API key. Get a new one from the list [here](https://docs.google.com/spreadsheets/d/1y8jZMQUjHh8jyyOXMIedEkw03IRoxUIGy88d58782WY)
+	- Verify your change has deployed before moving on: `curl -I https://ft-next-config-vars.herokuapp.com/app/YOUR_APP_NAME?apiKey=YOUR_API_KEY`
+	- Join the [ft-next-config-vars](https://dashboard.heroku.com/apps/ft-next-config-vars/resources) app in Heroku
+- Create a new application within Heroku correseponding to your apps name:
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: correseponding/corresponding

Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: apps/app's

File: docs/developer-guide/creating-new-apps.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+	- `heroku create YOUR_APP_NAME --region eu --org financial-times`
+	- Test your apps configuration running `nbt configure`
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: apps/app's

File: docs/developer-guide/creating-new-apps.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+- Configure your Travis CI build via the dashboard:
+	- Go to: https://travis-ci.org/Financial-Times/YOUR_APP_GITHUB_NAME/settings 
+	- Click on envinorment variables tab and add the following defaults:
```
Comment (keirog) [inline docs/developer-guide/creating-new-apps.md]: envinorment/environment

File: docs/developer-guide/troubleshooting.md [hunk 15]
```diff
@@ -21,3 +21,4 @@ permalink: /docs/developer-guide/troubleshooting/
   Then run `source ~/.bash_profile` to add it to your current terminal session
 - If bower complains of dependency resolution problems (or `make install` fails repeatedly at the bower step) run `rm -Rf bower_components` and try installing again. Avoid, if at all possible, manually resolving dependencies when bower gives you the option to.
 - If sass won't build (it may complain about too many open files) run `rm -Rf .sass-cache`
+- Are setting up a new app within Heroku or Travis? Read up more details on [creating new apps](/docs/developer-guide/creating-new-apps/) within the stack.
```
Comment (keirog) [inline docs/developer-guide/troubleshooting.md]: Are/Are you

**Comment by keirog** [comment]
(Sorry)

**Comment by phamann** [comment]
@keirog thanks :smile:

**Comment by keirog** [comment]
:+1:

**Comment by commuterjoy** [comment]
navigation *was* alphabetically ordered, ahem :)

**Pull request merged**